### ISSUE TITLE
Support aud as plain string instead of array of string (as configurable option)

### ISSUE THREAD

**Issue by mariotoffia** [lestrrat-go/jwx#368]
**Abstract**
Hi and thanks for a very nice library! :)

I'm working with [AWS Cognito "disability" of handling other than strings as values in a JWT](https://aws.amazon.com/premiumsupport/knowledge-center/decode-verify-cognito-json-token/). I'm wondering if it is possible for this library to accept a string, in addition to, the array of strings. 

Also have a configuration option when Marshal the token to produce a string instead of an array.

Cheers,
 Mario :)

**Describe the proposed solution/change**

For marshal:

Consider this
```go
        t := openid.New()
        t.Set(jwt.AudienceKey, `Golang Users`)

        buf, _ := json.MarshalIndent(t, "", "  ")
        fmt.Printf("%s\n", buf)
```

always produces

```json
{
  "aud": [
    "Golang Users"
  ]
}
```

If one may instead have an option on the token itself  to allow for variation when marshal. E.g. 
```go
        t := openid.New()
        t.Config(jwt.ConfigRenderAudStringWhenSingleEntry, true)
        t.Set(jwt.AudienceKey, `Golang Users`)

        buf, _ := json.MarshalIndent(t, "", "  ")
        fmt.Printf("%s\n", buf)
```

could then produce
```json
{
  "aud": "Golang Users"
}
```

Hence, conform to _AWS Cognito JWT_ ...

**Analysis**

N/A

**Additional context**

N/A

**Comment by lestrrat** [comment]
Thank you. I think the saner way would be to just default to a string "aud" when `len(t.audience) == 0`?

I wish there was a way to pass options to `"encoding/json".Marshal".... I really don't want to put options in EACH object.

**Comment by lestrrat** [comment]
@mariotoffia Wanna try this? https://github.com/lestrrat-go/jwx/tree/topic/single-aud

**Comment by mariotoffia** [comment]
@lestrrat thanks!! For me this is **spot on**!

Cheers,
 Mario :)

**Comment by lestrrat** [comment]
@mariotoffia Cool!

I think what I'm going to do is to make this behavior optional via a global setting. Something like

```
jwx.EncoderSettings(
   WithFlattenedJWTAudienceField(true), // Actual name TBD
)
```

This is because we may inadvertently change how the signatures are generated when somebody upgrades the library. This way its opt-in.

**Comment by lestrrat** [comment]
@mariotoffia Alright, I've settled on

```
jwt.Settings(jwt.WithFlattenAudience(true))
```

to trigger the desired behavior. Please check #369, and let me know what you think.

**Comment by mariotoffia** [comment]
@lestrrat This is perfect!

Cheers,
 Mario :)

**Comment by lestrrat** [comment]
fixed in #369

**Comment by lestrrat** [comment]
@mariotoffia merged, and I pushed a tag v1.1.8-rc1 so it's easier to reference it.

**Comment by mariotoffia** [comment]
Thanks for your kind support! :)
